src/ring_theory/polynomial/selmer.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2022 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+
+import data.polynomial.unit_trinomial
+import tactic.linear_combination
+
+/-!
+# Irreducibility of Selmer Polynomials
+
+This file proves irreducibility of the Selmer polynomials `X ^ n - X - 1`.
+
+## Main results
+
+- `polynomial.selmer_irreducible`: The Selmer polynomials `X ^ n - X - 1` are irreducible.
+
+TODO: Show that the Selmer polynomials have full Galois group.
+-/
+
+namespace polynomial
+open_locale polynomial
+
+lemma selmer_irreducible_aux (n : ℕ) (z : ℂ) : ¬ (z ^ n = z + 1 ∧ z ^ n + z ^ 2 = 0) :=
+begin
+  rintros ⟨h1, h2⟩,
+  replace h3 : z ^ 3 = 1,
+  { linear_combination (1 - z - z ^ 2 - z ^ n) * h1 + (z ^ n - 2) * h2 }, -- thanks polyrith!
+  have key : z ^ n = 1 ∨ z ^ n = z ∨ z ^ n = z ^ 2,
+  { rw [←nat.mod_add_div n 3, pow_add, pow_mul, h3, one_pow, mul_one],
+    have : n % 3 < 3 := nat.mod_lt n zero_lt_three,
+    interval_cases n % 3; simp only [h, pow_zero, pow_one, eq_self_iff_true, or_true, true_or] },
+  have z_ne_zero : z ≠ 0 :=
+  λ h, zero_ne_one ((zero_pow zero_lt_three).symm.trans (show (0 : ℂ) ^ 3 = 1, from h ▸ h3)),
+  rcases key with key | key | key,
+  { exact z_ne_zero (by rwa [key, self_eq_add_left] at h1) },
+  { exact one_ne_zero (by rwa [key, self_eq_add_right] at h1) },
+  { exact z_ne_zero (pow_eq_zero (by rwa [key, add_self_eq_zero] at h2)) },
+end
+
+lemma selmer_irreducible {n : ℕ} (hn1 : n ≠ 1) : irreducible (X ^ n - X - 1 : ℤ[X]) :=
```

Comment:
> 
```suggestion
lemma X_pow_sub_X_sub_one_irreducible {n : ℕ} (hn1 : n ≠ 1) : irreducible (X ^ n - X - 1 : ℤ[X]) :=
```
We try to use descriptive names even when they look long.

Reply:
Fair enough, especially since these lemmas won't be seeing much use.